projects/js-packages/publicize-components/src/components/connection-toggle/index.jsx

```diff
@@ -1,33 +1,25 @@
-import { FormToggle } from '@wordpress/components';
 import classnames from 'classnames';
 import PropTypes from 'prop-types';
 import ConnectionIcon from '../connection-icon';
 
 import './style.scss';
 
 const ConnectionToggle = props => {
-	const { className, checked, id, disabled, onChange, serviceName, label, profilePicture } = props;
-
+	const { checked, disabled, onChange, serviceName, label, profilePicture } = props;
 	const wrapperClasses = classnames( 'components-connection-toggle', {
 		'is-not-checked': ! checked,
 		'is-disabled': disabled,
 	} );
 
 	return (
-		<div className={ wrapperClasses }>
+		<div className={ wrapperClasses } title={ label }>
 			<ConnectionIcon
-				id={ id }
-				serviceName={ serviceName }
+				checked={ checked }
 				label={ label }
+				onClick={ onChange }
+				serviceName={ serviceName }
 				profilePicture={ profilePicture }
 			/>
-			<FormToggle
-				id={ id }
-				className={ className }
-				checked={ checked }
-				onChange={ onChange }
-				disabled={ disabled }
-			/>
```

Comment:
@pablinos This is where that bug of connections not disabled when limits are reached was introduced.